packages/block-library/src/gallery/save.js

```diff
@@ -17,14 +17,23 @@ export default function save( { attributes } ) {
 
 				switch ( linkTo ) {
 					case 'media':
-						href = image.url;
+						href = image.fullUrl || image.url;
```

Comment:
In what cases is there no full size?

Reply:
Current galleries don't have the fullUrl attribute set, this is a simple way to be back-compatible in these cases. Another even more important case where this may happen is when the gallery links to external images e.g: a set of image blocks linking to external URL's is transformed into a gallery.